sdk/core/Azure.Core/tests/TestFramework/ClientDiagnosticListener.cs

```diff
@@ -56,9 +56,11 @@ public void OnNext(KeyValuePair<string, object> value)
                 else if (value.Key.EndsWith(stopSuffix))
                 {
                     var name = value.Key.Substring(0, value.Key.Length - stopSuffix.Length);
+                    PropertyInfo propertyInfo = value.Value.GetType().GetProperty("Id");
```

Comment:
Would using the activity reference work here?

Reply:
That appears to work and is much better!